src/neo-vm/ExecutionEngine.cs

```diff
@@ -872,26 +931,64 @@ private bool ExecuteInstruction()
                         }
                         break;
                     }
-                case OpCode.PACK:
+                case OpCode.HASKEY:
                     {
-                        if (!TryPop(out int size)) return false;
-                        if (size < 0 || size > context.EvaluationStack.Count)
-                            return false;
-                        VMArray array = new VMArray(ReferenceCounter);
-                        for (int i = 0; i < size; i++)
+                        if (!TryPop(out PrimitiveType key)) return false;
+                        if (!TryPop(out StackItem x)) return false;
+                        switch (x)
                         {
-                            if (!TryPop(out StackItem item)) return false;
-                            array.Add(item);
+                            case VMArray array:
+                                {
+                                    int index = key.ToInt32();
+                                    if (index < 0) return false;
+                                    Push(index < array.Count);
+                                    break;
+                                }
+                            case Map map:
+                                {
+                                    Push(map.ContainsKey(key));
+                                    break;
+                                }
+                            case Buffer buffer:
```

Comment:
ByteArray too?

Reply:
Now we treat `ByteArray` as an immutable `string`.

Reply:
But maybe is interesting to use if we want to know if the length is less than `index`. I think that the use case could be the same as Buffer for `HASKEY`

Reply:
Done.